docs/code-samples/lifecycling/daml/Scripts/FloatingRateBond.daml

```diff
@@ -0,0 +1,182 @@
+module Scripts.FloatingRateBond where
+
+import DA.Date (Month(..), date)
+import DA.Map qualified as M (empty, fromList)
+import DA.Set qualified as S (singleton)
+import Daml.Script
+
+-- INTERFACE DEPENDENCIES --
+import Daml.Finance.Interface.Instrument.Bond.FloatingRate.Factory qualified as FloatingRate (Create(..), F(..))
+import Daml.Finance.Interface.Instrument.Bond.FloatingRate.Types (FloatingRate(..))
+import Daml.Finance.Interface.Lifecycle.Rule.Claim qualified as Claim (ClaimEffect(..))
+import Daml.Finance.Interface.Lifecycle.Rule.Lifecycle qualified as Lifecycle (Evolve(..))
+import Daml.Finance.Interface.Settlement.Batch qualified as Batch (Settle(..))
+import Daml.Finance.Interface.Settlement.Instruction qualified as Instruction (Allocate(..), Approve(..))
+import Daml.Finance.Interface.Settlement.Types (Allocation(..), Approval(..))
+import Daml.Finance.Interface.Types.Common.Types (Id(..), InstrumentKey(..))
+import Daml.Finance.Interface.Types.Date.Calendar (BusinessDayConventionEnum(..))
+import Daml.Finance.Interface.Types.Date.DayCount (DayCountConventionEnum(..))
+import Daml.Finance.Interface.Types.Date.RollConvention (PeriodEnum(..))
+
+-- IMPLEMENTATION DEPENDENCIES --
+import Daml.Finance.Data.Numeric.Observation (Observation(..))
+import Daml.Finance.Data.Time.DateClock (dateToDateClockTime)
+import Daml.Finance.Instrument.Bond.FloatingRate.Factory qualified as FloatingRate (Factory(..))
+
+import Workflow.CreditAccount qualified as CreditAccount
+
+import Scripts.FixedRateBond (FixedRateState(..), runFixedRateBond)
+
+-- | Test script that
+-- 1. creates a floating rate bond instrument and a holding on it
+-- 2. reuses the lifecycle rule and lifecycle event from the fixed rate bond tutorial
+-- 3. lifecycles the time event
+-- 4. processes the lifecycle effect
+-- 5. settles the distribution
+runFloatingRateBond : Script()
+runFloatingRateBond = do
+
+  -- Execute the `runFixedRateBond` script to reuse variables.
+  FixedRateState{bank
+    , bob
+    , public
+    , bobAccount
+    , periodicSchedule
+    , usdInstrument
+    , holidayCalendarIds
+    , lifecycleRuleCid
+    , lifecycleClaimRuleCid
+    , firstCouponClockEventCid} <- runFixedRateBond
+
+  now <- getTime
+
+  let pp = [("Public Party", S.singleton public)]
+
+  -- Create and distribute bond
+  -- CREATE_FLOATING_RATE_BOND_VARIABLES_BEGIN
+  let
+    notional = 1.0
+    referenceRateId = "USD/LIBOR/3M"
+    couponSpread = 0.011
+    couponPeriod = M
+    couponPeriodMultiplier = 12
+    dayCountConvention = ActActISDA
+    businessDayConvention = Following
+  -- CREATE_FLOATING_RATE_BOND_VARIABLES_END
+
+  -- CREATE_FLOATING_RATE_OBSERVATIONS_BEGIN
+  let
```

Comment:
It feels more realistic to create the observation after the creation of the instrument

Reply:
Thanks! Btw, do we need both observations or is one sufficient?

Reply:
Currently, only the first one is needed, but I wasn't sure how far to lifecycle the instrument in the tutorial. In order to lifecycle the final coupon/redemption, the second one would be needed as well, so I kept it in there for now.

Reply:
When you lifecycle the first coupon, the second observation is in principle not known, so I would remove it from the example, wdyt?

Reply:
Sure, makes sense. I removed it now.